scylla/src/transport/connection.rs

```diff
@@ -124,6 +133,36 @@ impl Connection {
         Ok((connection, error_receiver))
     }
 
+    #[cfg(feature = "ssl")]
+    async fn get_worker_handle(
```

Comment:
What about changing name of `get_worker_handle` to `run_router` and `run_router` to `run_router_aux`? Merging three of these functions (`get_worker_handle` has two variants) would create two unnecessary big function with identical instructions at the end.

Also I'm not sure but there might be some type error if we try to put choosing from config if the TcpStream should be transformed into SslStream, because it was the reason why we had to make `run_router` separate from constructor and `router` function.

Reply:
> What about changing name of `get_worker_handle` to `run_router` and `run_router` to `run_router_aux`? Merging three of these functions (`get_worker_handle` has two variants) would create two unnecessary big function with identical instructions at the end.

OK, I see. I tried inlining `run_router` into `get_worker_handle` myself and I changed my mind :) While it compiled successfully, code repetition looked more jarring than I thought. I'm OK with how the functions are separated now. I'd rather see a more descriptive name than `run_router_aux`, but myself I don't have a better suggestion, so I think that your suggestion for rename is fine.